packages/plugins/experimental/plugin-script/src/templates/queue.ts

```diff
@@ -0,0 +1,149 @@
+//
+// Copyright 2025 DXOS.org
+//
+/* eslint-disable no-console */
+// TODO(ZaymonFC): Make dxos log available.
+
+// @ts-ignore
+import { createStatic, defineFunction, S, ObjectId, EchoObject, Filter } from 'dxos:functions';
+// @ts-ignore
+import { FetchHttpClient } from 'https://esm.sh/@effect/platform@0.77.2?deps=effect@3.13.3';
+// @ts-ignore
+import { Effect } from 'https://esm.sh/effect@3.13.3';
+
+export default defineFunction({
+  inputSchema: S.Any,
+  outputSchema: S.Struct({ success: S.Boolean }),
+  handler: ({ event, context: { space } }: any) =>
+    Effect.gen(function* () {
+      console.log('Starting queue function', JSON.stringify(event.data));
+
+      console.log('Upserting contact stored schema');
+      const [contactStoredSchema] = yield* Effect.tryPromise({
+        try: () => space.db.schemaRegistry.register([ContactType]),
+        catch: (e: any) => e,
+      });
+
+      const item = event.data.item as MessageType;
+      console.log('Found item', JSON.stringify(item, null, 2));
+
+      const { objects } = yield* Effect.tryPromise({
+        try: () => space.db.query(Filter.schema(contactStoredSchema)).run(),
+        catch: (e: any) => e,
+      });
+      console.log('Found contacts', objects);
+
+      const sender = (item as any).sender;
+
+      const contactExists = (objects as ContactType[]).findIndex((c) => c.email === sender.email) !== -1;
+      if (contactExists) {
+        console.log('Contact already exists. No action required.');
+        return { success: true };
+      }
+
+      const contact = createStatic(contactStoredSchema, { name: sender.name, email: sender.email });
+
+      const addedObject = space.db.add(contact);
+      console.log('Added contact to space', JSON.stringify(addedObject));
+
+      yield* Effect.tryPromise({
+        try: () => space.db.flush(),
+        catch: (e: any) => e,
+      });
+
+      return { success: true };
+    }).pipe(Effect.provide(FetchHttpClient.layer)),
+});
+
+//
+// Types
+//
+
+const ContactType = S.Struct({
+  name: S.optional(S.String),
+  email: S.optional(S.String),
+}).pipe(EchoObject('example.org/TempContactType', '0.1.0'));
+type ContactType = S.Schema.Type<typeof ContactType>;
+
+const ActorRoles = ['user', 'assistant'] as const;
```

Comment:
probably remove all these unused schemas

Reply:
These are all required to type the incoming data as `MessageType`

Reply:
ah I missed that cast